src/utils/plotSpecs/generateMitochondrialContentSpec.js

```diff
@@ -139,4 +139,35 @@ const generateSpec = (config, plotData) => {
   };
 };
 
-export default generateSpec;
+const generateData = (
+  doubletScore,
+  selectedSample,
+  embeddingData,
+  cellSetProperties,
+) => {
+  // the result of the map on embedding data contains non-consecutive indices (aka empties)
+  // filter removes missing elements in the array thus effectively reindexing the resulting array
+  // into consecutive indexes which is the format expected by Vega.
+  // If we have a specific sample then filter will act normally, in case of using all samples
+  // we just provide a dummy filter function.
+  // See for more information: https://262.ecma-international.org/5.1/#sec-15.4.4.20
+  let filterFunc = () => true;
+  if (selectedSample !== 'All') {
+    const cellIds = Array.from(cellSetProperties[selectedSample].cellIds);
+    filterFunc = (val, idx) => cellIds.includes(idx);
+  }
+  const filteredData = embeddingData
+    .map((coord, cellId) => ({
+      x: coord[0],
+      y: coord[1],
+      'mt-content': doubletScore[cellId],
+    }))
+    .filter(filterFunc);
+
+  return filteredData;
+};
```

Comment:
this is essentially almost the same piece of code like the one in `src/utils/plotSpecs/generateDoubletScoresSpec.js` -- have you considered moving it to a separate function that you can reuse in both places?

Reply:
These 2 plots, `configureEmbedding` and `doubletScore` plot are actually just a variation of `embeddingContinuousPlot`. Ideally, they should be reusable, like how `categoricalEmbedding` is reused in this step. I guess no one wants to take the time to refactor these plots yet XD. I'll take a look at them

Reply:
thanks Agi, that would be awesome!

Reply:
Refactored into one plot in 39ef247